lib/active_publisher/async/in_memory_adapter/consumer_thread.rb

```diff
@@ -102,13 +101,15 @@ def start_consuming_thread
             # If the queue is empty, we should continue to update to "last_tick_at" time.
             next if current_messages.nil?
 
+            @channel ||= make_channel
```

Comment:
Will this stop publishing from happening during tests or only defer the error? Will this cause some background thread to  die?

Reply:
No, it matches what the old code did before your PR.  It just delays created the channel until there is at least one message.  So if any tests actually publish messages, then the channel will still get created and used.

Note that Grunt currently doesn't actually publish any messages in its specs, nor does have it rabbitmq as a requirement to be installed on the CI or anything.  I don't see any reason to break grunt or any other projects by forcing the channel to be created upfront.  So I'm just restoring the previous behavior here.

I'm not sure what this would have to do with causing any background threads to die.  Can you elaborate on that?

Reply:
That makes sense. I forgot that I moved that line up.